proto/xudrpc.proto

```diff
@@ -141,15 +141,22 @@ service Xud {
     };
   }
   
-  /* Subscribe to peer order events. */
-  rpc SubscribePeerOrders(SubscribePeerOrdersRequest) returns (stream SubscribePeerOrdersResponse) {
+  /* Subscribe to orders being added to the order book. */
+  rpc SubscribeAddedOrders(SubscribeAddedOrdersRequest) returns (stream Order) {
     option (google.api.http) = {
-      get: "/v1/subscribepeerorders"
+      get: "/v1/subscribeorders"
     };
   }
   
-  /* Subscribe executed swaps. */
-  rpc SubscribeSwaps(SubscribeSwapsRequest) returns (stream SubscribeSwapsResponse) {
+  /* Subscribe to orders being removed from the order book. */
+  rpc SubscribeRemovedOrders(SubscribeRemovedOrdersRequest) returns (stream OrderRemoval) {
```

Comment:
Segregating on removed/added instead of ownOrder/peerOrder is fine, but less optimal IMO. 
I already [specified](https://github.com/ExchangeUnion/xud/pull/500#issuecomment-423241655) the benefits of the later.
 The former seems to have the same problem - ownOrder and peerOrder structures are not equal, and the current `Order` protobuf type contains them both. I would actually like to see a separate `OwnOrder` + `PeerOrder` protobuf types. It would be useful for `placeOrder` response. 

If order removal events are missing some fields, which are actually not necessary, we can either retrieve them (if applicable) and emit anyway, or not populate them in the response (same like `peerPubKey` is not populated for ownOrder, and `localId` for peerOrder).  If i'm missing some technical challenge here, let me know.

In addition, relying on `localId` for differentiation is problematic. It's not very self evident, and will raise issues for #223.

With ownOrder/peerOrder segregation we'll have the same problem with quantity, but the positive/negative solution that you suggested sounds like a good solution.

Reply:
In regards to the `Order` protobuf and possibly splitting the types based on own or peer order, I actually think a [oneof](https://developers.google.com/protocol-buffers/docs/proto3#oneof) field fits perfectly. Their structures are identical except for a pair of fields, only one of which applies to each type. That's also why I feel the a single message type can be cleanly used for the added orders. Also I think splitting them would complicate the `getOrders` call - unless we went back to splitting them into `repeated own_orders` and `repeated peer_orders` which I don't think is ideal.

As for whether relying on `localId` is problematic, it doesn't seem like a big deal to me however I'm only basing that on my own intuition and maybe it really isn't self evident enough (although I would certainly document it as best I can). To me as well, it's the same sort of logic that's behind peer/own orders being combined in `getOrders` for instance, where clients can also infer which orders are local by the `localId`. Maybe we could use another field to make this clearer? `is_local`? Regarding #223, my most recent thoughts there weren't necessarily to have`localId` undefined or removed but rather just set it equal to `orderId`, so this wouldn't necessarily conflict although it would limit our options.

It's definitely possible to have the order removals use the `Order` message type, leaving the non-applicable fields like `price` with default values and using negative values for quantity. That's what I was going for at first, and while there wasn't a technical challenge really I did feel like the logic started to get messy when it came to dealing with `StampedOwnOrder | OrderPortion` variables passed to the GrpcService callback. From purely the protobuf message standpoint I don't think it's actually a big deal to leave a handful of fields unused.

I guess some key questions to me are:

- Are we generally comfortable with the concept of combining peer & own orders on the rpc layer? If not, maybe we should revisit `getOrders`?

- Is a field like `is_local` on orders something we'd want, perhaps even useful for the `getOrders` call without consideration for these new calls? Vs the alternative of checking for a value on `local_id`.

Thanks for all the thought you've put into this.

Reply:
I wasn’t aware of this `oneof`, looks like it can work well. 

In this regard, I don’t see a technical issue with combining peerOrder & ownOrder. 
I’m more inclined towards separation mostly for the purpose of “educating” the client. I think that the concept of ownOrder/peerOrder is weird and uncommon, so having it clearly defined in the API as much as possible will help users/developers to understand what is xud about. 
Having it merely mentioned in the documentation won't have the same effect. 
This separation will also lead to better understanding of `placeOrder` response, which isn't trivial.
We should definitely not decide against it because of the current `getOrders` implementation, which is of minor importance. 

In regards to having #223 with `localId` equals to the global id - It can work technically, but i’m not sure it’s the best solution. 

in regards to having a field like `is_local` - this name isn’t so self explanatory as well IMO. if we’ll do that, perhaps simply use `is_own_order`. So I'd say it's either we'll use an extra field like that, or separate the methods/types for ownOrder/peerOrder instead. Feel free to decide about this yourself, or wait for @kilrau @michael1011 input as well.

Reply:
`is_own_order` I agree is more clear. I think that might be the way to go, it would also be a nice enhancement to `getOrders` I think. That all seems clear to enough to me. I will write up detailed documentation tonight and then we can see if it makes sense.